fs/span-manager/span_manager.go

```diff
@@ -409,8 +409,8 @@ func (m *SpanManager) addSpanToCache(spanID compression.SpanID, contents []byte,
 // getSpanFromCache returns the cached span content as an `io.Reader`.
 // `offset` is the offset of the requested contents within the span.
 // `size` is the size of the requested contents.
-func (m *SpanManager) getSpanFromCache(spanID compression.SpanID, offset, size compression.Offset) (io.ReadCloser, error) {
-	rc, err := m.cache.Get(fmt.Sprintf("%d", spanID))
+func (m *SpanManager) getSpanFromCache(spanID compression.SpanID, offset, size compression.Offset, opts ...cache.Option) (io.ReadCloser, error) {
+	rc, err := m.cache.Get(fmt.Sprintf("%d", spanID), opts...)
```

Comment:
nit: Since this is only called within the same object can we avoid mutating the func and instead pass `m.opts` here?

Reply:
Sure. I'll fix `addSpanToCache` too.